code/modules/vehicles/spacepod/ui.dm

```diff
@@ -0,0 +1,80 @@
+// credits to interception for UI
+/datum/action/vehicle/sealed/pod_status
+	name = "Control Panel"
+	desc = "See an overview of the pod."
+	background_icon_state = "bg_tech"
+	overlay_icon_state = "bg_tech_border"
+	button_icon = 'icons/obj/tram/tram_controllers.dmi'
+	button_icon_state = "tram-controller"
+
+/datum/action/vehicle/sealed/pod_status/Trigger(trigger_flags)
+	. = ..()
+	vehicle_entered_target.ui_interact(owner)
+
+/obj/vehicle/sealed/space_pod/ui_data(mob/user)
+	. = list(
+		"name" = name,
+		"power" = isnull(cell) ? 0 : cell.charge,
+		"maxPower" = isnull(cell) ? 0 : cell.maxcharge,
+		"health" = get_integrity_percentage(),
+		"acceleration" = !isnull(drift_handler) ? drift_handler.drift_force : 0,
+		"maxAcceleration" = max_speed,
+		"forcePerMove" = force_per_move,
+		"headlightsEnabled" = light_on,
+		"cabinPressure" = !isnull(cabin_air_tank) ? "[round(cabin_air.return_pressure(), 0.1)]kPa" : "No air tank",
+		"occupantcount" = length(occupants),
+		"occupantmax" = max_occupants,
+		"partUIData" = list(),
+	)
+	for(var/obj/item/pod_equipment/equipment as anything in get_all_parts())
+		if(!equipment.interface_id)
+			continue
+		var/part_ref = REF(equipment)
+		.["partUIData"][part_ref] = equipment.ui_data(user)
+		.["partUIData"][part_ref]["ref"] = part_ref
+
+/obj/vehicle/sealed/space_pod/ui_static_data(mob/user)
+	. = list(
+		"parts" = list(),
+	)
+	for(var/obj/item/pod_equipment/equipment as anything in get_all_parts())
+		var/list/info = list()
+		info["name"] = equipment.name
+		info["desc"] = equipment.desc
+		info["type"] = equipment.interface_id
+		info["ref"] = REF(equipment)
+		.["parts"] += list(info)
+
+/obj/vehicle/sealed/space_pod/ui_status(mob/living/user, datum/ui_state/state)
+	if(!(user in occupants) || !(occupants[user] & VEHICLE_CONTROL_DRIVE))
```

Comment:
this in check is redundant

Reply:
you can still have the ui open in an uninteractive while near the pod which would make the rightmost part runtime

Reply:
why would it runtime? we'd do list[not_in_list] which is just null, and null & flag which is 0